src/Test/Html/Selector.elm

```diff
@@ -186,68 +189,109 @@ id =
 -}
 tag : String -> Selector
 tag name =
-    Tag
-        { name = name
-        , asString = "tag " ++ toString name
-        }
+    Tag name
 
 
-{-| Matches elements that have the given attribute with the given string value.
+{-| Matches elements that have the given attribute in a way that makes sense
+given their semantics in `Html`.
 
-For attributes with boolean values, such as `checked`, use [`boolAttribute`](#boolAttribute).
-
-    import Html
-    import Html.Attributes as Attr
-    import Test.Html.Query as Query
-    import Test exposing (test)
-    import Test.Html.Selector exposing (attribute, text)
 
+#### How matching works:
 
-    test "the welcome <h1> says hello!" <|
-        \() ->
-            Html.div []
-                [ Html.h1 [ Attr.title "greeting" ] [ Html.text "Hello!" ] ]
-                |> Query.fromHtml
-                |> Query.find [ attribute "title" "greeting" ]
-                |> Query.has [ text "Hello!" ]
+  - `Html.Attributes.class` and `Html.Attributes.classList` will work the same as
+    `Selectors.classes`, matching any element with at least the given classes.
 
--}
-attribute : String -> String -> Selector
-attribute name value =
-    Attribute
-        { name = name
-        , value = value
-        , asString = "attribute " ++ toString name ++ " " ++ toString value
-        }
+  - `Html.Attributes.style` will work the same way as `Selectors.styles`, matching
+    any element with at least the given style properties.
 
+  - Any other `String` attributes like `title` or `Bool` attributes like
+    `disabled` will match elements with the exact value for those attributes.
 
-{-| Matches elements that have the given attribute with the given boolean value.
+  - Any attributes from `Html.Events`, or attributes with values that have types
+    other than `String` or `Bool` will not match anything.
 
-For attributes with string values, such as `title`, use [`attribute`](#attribute).
+The example below demonstrates usage
```

Comment:
This documentation is way too long to fit nicely on the package site -- better to split it out into the README, and link to it instead for "a fully example"